### ISSUE TITLE
end-to-end sfm

### ISSUE THREAD

**Issue by jdk9405** [facebookresearch/vggsfm#15]
Hi! Thanks for sharing your amazing work!

I'm interested in the end-to-end pipeline for SfM as mentioned in the paper, but I could not find the usage of Theseus library in this code.

Is there a particular reason to use pycolmap in BA instead of Theseus?

**Comment by jytime** [comment]
Hi Dong,

The currently released code only supports inference. We used pycolmap in BA during inference because it is 2.5~3 times faster than Theseus in practice. For inference, we don't need gradients so we can use any type of BA implementations.

We plan to provide the code for Theseus BA when we gradually release the code for training (sry cannot guarantee a timeline for it). However, if you want to try it yourself, here is a good example:

https://github.com/facebookresearch/theseus/blob/9a117fd02867c5007c6686e342630f110e488c65/theseus/utils/examples/bundle_adjustment/data.py#L151

You just need to modify our BA code to the Theseus format as above. Remember to reduce the number of tracks to save GPU memory.

**Comment by jdk9405** [comment]
I understood. Thank you for the prompt response, and I await the release of the training code! :)
